httplib.h

```diff
@@ -7834,10 +7898,11 @@ inline bool SSLClient::load_certs() {
         ret = false;
       }
     } else {
+      SSL_CTX_set_default_verify_paths(ctx_);
```

Comment:
This call should be in `#else` block. Please see the original code.

Reply:
I am not very familiar with openssl, but I have come to the following. 
If we fail to load certs from the system storage, then it would be correct to call SSL_CTX_set_default_verify_paths. Am I wrong?

Reply:
On Windows maybe it didn't matter, but for MacOS it has value at all I guess…

Reply:
Here is the link to `SSL_CTX_set_default_verify_paths`:
https://www.openssl.org/docs/man3.1/man3/SSL_CTX_set_default_verify_paths.html
<img width="811" alt="image" src="https://user-images.githubusercontent.com/357397/218859763-4d7a8b9e-68fe-4339-a908-b730b5581814.png">
So if we successfully load the OS certs, we shouldn't call `SSL_CTX_set_default_verify_paths`.

You probably could do as below:

```cpp
      auto loaded = false;
#ifdef _WIN32
      loaded = detail::load_system_certs_on_windows(SSL_CTX_get_cert_store(ctx_));
#elif defined(__APPLE__)
      loaded = detail::load_system_certs_on_apple(SSL_CTX_get_cert_store(ctx_));
#endif
      if (!loaded) { SSL_CTX_set_default_verify_paths(ctx_); }
```

This means, always `SSL_CTX_set_default_verify_paths` gets called on linux. As for Windows and MacOS, if the `detail::load_system_certs_on_???` fails, then `SSL_CTX_set_default_verify_paths` will be called as the last resort function.

Reply:
Yeah, you're right. But when we first use SSL_CTX_set_default_verify_paths, it means that if something screws up, the system will at least try to find certificates in the default places. As far as I can tell, adding extra certificates using X509_STORE_add_cert after setting the default paths shouldn't cause any issues.

Reply:
To put it another way, I believe that calling SSL_CTX_set_default_verify_paths first and then load_system_certs will either result in the same or an even better state than if load_system_certs were called first and then set_default_verify_paths.

Reply:
In the original code for Windows, we don't have to call `SSL_CTX_set_default_verify_paths` before/after `load_system_certs_on_windows`. I simply would like to do the same for MacOS since it is absolutely unnecessary to call `SSL_CTX_set_default_verify_paths` if `load_system_certs_on_apple` succeeds.

Reply:
ok, done

Reply:
If you really want to set the default path before loading certs, just call it after initializing `httplib::SSLClient`.

```cpp
httplib::SSLClient client;
SSL_CTX_set_default_verify_paths(client.ssl_context());
```

Reply:
If you really want to set the default path before loading certs, just call it after initializing httplib::SSLClient.

```cpp
httplib::SSLClient client;
SSL_CTX_set_default_verify_paths(client.ssl_context());
```